services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -546,31 +569,65 @@ getConferenceCallingInternal (Right tid) = do
   Members '[GundeckAccess, TeamFeatureStore, TeamStore, P.TinyLog] r =>
   TeamId ->
-  Public.TeamFeatureStatus 'Public.TeamFeatureConferenceCalling ->
-  Sem r (Public.TeamFeatureStatus 'Public.TeamFeatureConferenceCalling)
+  Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureConferenceCalling ->
+  Sem r (Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureConferenceCalling)
 setConferenceCallingInternal =
   setFeatureStatusNoConfig @'Public.TeamFeatureConferenceCalling $ \_status _tid -> pure ()
 
 getSelfDeletingMessagesInternal ::
   Member TeamFeatureStore r =>
   GetFeatureInternalParam ->
-  Sem r (Public.TeamFeatureStatus 'Public.TeamFeatureSelfDeletingMessages)
+  Sem r (Public.TeamFeatureStatus 'Public.WithPaymentStatus 'Public.TeamFeatureSelfDeletingMessages)
 getSelfDeletingMessagesInternal = \case
   Left _ -> pure Public.defaultSelfDeletingMessagesStatus
-  Right tid ->
-    TeamFeatures.getSelfDeletingMessagesStatus tid
-      <&> maybe Public.defaultSelfDeletingMessagesStatus id
+  Right tid -> do
+    (maybeFeatureStatus, maybePaymentStatus) <- TeamFeatures.getSelfDeletingMessagesStatus tid
+    pure $ case (maybePaymentStatus, maybeFeatureStatus) of
+      (Just Public.PaymentUnlocked, Just featureStatus) ->
+        Public.TeamFeatureStatusWithConfigAndPaymentStatus
+          (Public.tfwcStatus featureStatus)
+          (Public.tfwcConfig featureStatus)
+          Public.PaymentUnlocked
+      (Just Public.PaymentUnlocked, Nothing) ->
+        Public.TeamFeatureStatusWithConfigAndPaymentStatus
+          (Public.tfwcapsStatus Public.defaultSelfDeletingMessagesStatus)
+          (Public.tfwcapsConfig Public.defaultSelfDeletingMessagesStatus)
+          Public.PaymentUnlocked
+      _ -> Public.defaultSelfDeletingMessagesStatus
 
 setSelfDeletingMessagesInternal ::
-  Members '[GundeckAccess, TeamFeatureStore, TeamStore, P.TinyLog] r =>
+  Members
+    '[ GundeckAccess,
+       TeamStore,
+       TeamFeatureStore,
+       P.TinyLog,
+       Error TeamFeatureError
+     ]
+    r =>
   TeamId ->
-  Public.TeamFeatureStatus 'Public.TeamFeatureSelfDeletingMessages ->
-  Sem r (Public.TeamFeatureStatus 'Public.TeamFeatureSelfDeletingMessages)
+  Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureSelfDeletingMessages ->
+  Sem r (Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureSelfDeletingMessages)
 setSelfDeletingMessagesInternal tid st = do
+  guardPaymentStatus @'Public.TeamFeatureSelfDeletingMessages tid (Public.PaymentStatus Public.PaymentLocked)
   let pushEvent =
         pushFeatureConfigEvent tid $
           Event.Event Event.Update Public.TeamFeatureSelfDeletingMessages (EdFeatureSelfDeletingMessagesChanged st)
-  (TeamFeatures.setSelfDeletingMessagesStatus tid st) <* pushEvent
+  TeamFeatures.setSelfDeletingMessagesStatus tid st <* pushEvent
+
+-- TODO(fisx): move this function to a more suitable place / module.
```

Comment:
where would you move this?  wire-api?  or keep it in this module?  i'm not sure.

Reply:
I don't think we can move this to `wire-api` because it accesses C*. If we move it, I think we would have to pass the payment status that comes from the db. Not sure, let's keep it here for now, and maybe later we can move this for all team feature in one place?